src/main/ruby/truffleruby/core/integer.rb

```diff
@@ -332,6 +332,19 @@ def zero?
     self == 0
   end
 
+  def self.try_convert(obj)
+    unless obj.respond_to?(:to_int)
+      return nil
+    end
+
+    result = obj.to_int
+    if result.nil? or result.is_a?(Integer)
```

Comment:
```suggestion
    if Primitive.nil?(result) or Primitive.object_kind_if?(result, Integer)
```

These two do the same but are not affected by redefinitions or overrides of `nil?/is_a?` which we wouldn't want here.